include/onnxruntime/core/session/onnxruntime_lite_custom_op.h

```diff
@@ -585,20 +806,49 @@
       return self->output_types_[indice];
     };
 
-    OrtCustomOp::GetInputCharacteristic = [](const OrtCustomOp*, size_t) {
-      return INPUT_OUTPUT_OPTIONAL;
+    OrtCustomOp::GetInputCharacteristic = [](const OrtCustomOp* op, size_t) {
+      auto self = reinterpret_cast<const OrtLiteCustomOp*>(op);
+      return (self->input_types_.empty() || self->input_types_[0] != ONNX_TENSOR_ELEMENT_DATA_TYPE_UNDEFINED) ? INPUT_OUTPUT_OPTIONAL : INPUT_OUTPUT_VARIADIC;
```

Comment:
>self->input_types_[0] != ONNX_TENSOR_ELEMENT_DATA_TYPE_UNDEFINE [](http://example.com/codeflow?start=44&length=63)

it looks weird to use "UNDEFINED" type for varadic. should we have explicit type defined?

Reply:
Because we want to allow variadic types in the "TensorArray".